source/core/src/main/com/csse3200/game/components/player/PlayerActions.java

```diff
@@ -28,20 +28,22 @@ public void create() {
     entity.getEvents().addListener("walk", this::walk);
     entity.getEvents().addListener("walkStop", this::stopWalking);
     entity.getEvents().addListener("attack", this::attack);
+    entity.getEvents().addListener("interact", this::interact);
   }
 
   @Override
   public void update() {
     if (moving) {
       updateSpeed();
     }
+    updateInteraction();
+  }
 
-    // Check for the closest sensor
+  private void updateInteraction() {
```

Comment:
Could add a docstring here to make it clear what the function is doing

Reply:
Thanks for the feedback however I think this function is self explanitory due to it's name and neighboring functions